src/StatsTools/reweighting.jl

```diff
@@ -394,18 +405,245 @@ function mixed_estimator_df_progress(shift::Vector, hproj::Vector, vproj::Vector
+```
+using data from multiple replicas. 
+
+Argument `op_ol` holds data for the operator overlap ``\\mathbf{c}_a^{(n)} \\hat{A} \\mathbf{c}_b^{(n)}`` 
+and `vec_ol` holds data for the vector overlap ``\\mathbf{c}_a^{(n)} \\mathbf{c}_b^{(n)}``.
+They are of type `Vector{Vector}`, with each element `Vector` 
+holding the data for a pair of replicas.
+Argument `shift` is of type `Vector{Vector}`, with each element `Vector` 
+holding the shift data for each individual replica.
+
+The reweighting is an extension of the mixed estimator using the reweighting technique 
+described in [Umrigar *et al.* (1993)](http://dx.doi.org/10.1063/1.465195). 
+Reweighting is done over `h` time steps and `length(shift) - skip` time steps are used 
+for the blocking analysis done with [`ratio_of_means()`](@ref). 
+`dτ` is the time step and `weights` a function that
+calulates the weights. See [`w_exp()`](@ref) and [`w_lin()`](@ref).
+Additional keyword arguments are passed on to [`ratio_of_means()`](@ref).
+
+Error propagation is done with [`MonteCarloMeasurements`](@ref).
+Results are returned as [`RatioBlockingResult`](@ref).
+
+See also [`mixed_estimator`](@ref), [`growth_estimator()`](@ref).
+"""
+function rayleigh_replica_estimator(
+    op_ol::Vector, 
+    vec_ol::Vector, 
+    shift::Vector, 
+    h, 
+    dτ;
+    skip=0,
+    E_r::Vector=[mean(view(s, skip+1:length(s))) for s in shift],
+    weights=w_exp,
+    kwargs...
+)   
+    num_reps = length(shift)
+    pair_idx = 0
+    num = zeros(length(op_ol[1]) - skip)
+    denom = zeros(length(op_ol[1]) - skip)
+    for a in 1:num_reps, b in a+1:num_reps
+        pair_idx += 1
+        wts = if h == 0
+            ones(length(num))
+        else
+            weights(shift[a], h, dτ; E_r=E_r[a], skip) .* weights(shift[b], h, dτ; E_r=E_r[b], skip)
+        end        
+        @views num += wts .* op_ol[pair_idx][skip+1:end]
+        @views denom += wts .* vec_ol[pair_idx][skip+1:end]
+    end
+    return ratio_of_means(num, denom; kwargs...)
+end
+"""
+    rayleigh_replica_estimator(
+        df::DataFrame;
+        op_ol="Op1", 
+        vec_ol="dot", 
+        skip=0, 
+        Anorm=1,
+        kwargs...
+    ) -> r::RatioBlockingResult
+Compute the estimator of a Rayleigh quotient of operator ``\\hat{A}`` 
+(without reweighting i.e. `h = 0`)
+```math
+A_\\mathrm{est} = \\frac{\\sum_{a<b} \\sum_n \\mathbf{c}_a^{(n)} \\cdot \\hat{A} \\cdot \\mathbf{c}_b^{(n)}}
+    {\\sum_{a<b} \\sum_n \\mathbf{c}_a^{(n)} \\cdot \\mathbf{c}_b^{(n)}},
+```
+directly from a `DataFrame` returned by [`lomc!`](@ref). 
+The keyword arguments `shift`, `op_ol` and `vec_ol` can be used to change the names of the relevant columns.
+The operator overlap data can be scaled by a prefactor `Anorm`.
+
+See [`AllOverlaps`](@ref).
+"""
+function rayleigh_replica_estimator(
+    df::DataFrame;
+    shift="shift",
+    op_ol="Op1", 
+    vec_ol="dot", 
+    h=0,
```

Comment:
Good to see that the option to set `h` has been added. The docstring still needs to be updated to reflect this change.